src/test/java/seedu/address/logic/commands/AddAppointmentCommandTest.java

```diff
@@ -0,0 +1,146 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.appointment.AddAppointmentCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.AppointmentTime;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.TypicalPersons;
+
+public class AddAppointmentCommandTest {
+
+    private Model model = new ModelManager(TypicalPersons.getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_appointmentAcceptedByModel_addAppointmentSuccessful() {
+        Person firstPerson = model.getAddressBook().getPersonList().get(0);
+        UUID personId = firstPerson.getId();
+        AppointmentTime appointmentTime = new AppointmentTime("10/04/2024 2PM-3PM");
+        Appointment appointmentToAdd = new Appointment(personId, appointmentTime);
+        AddAppointmentCommand addAppointmentCommand = new AddAppointmentCommand(INDEX_FIRST_PERSON, appointmentTime);
+
+        String expectedMessage = String.format(AddAppointmentCommand.MESSAGE_SUCCESS,
+                Messages.formatAppointment(appointmentToAdd, firstPerson.getName().fullName));
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.addAppointment(appointmentToAdd);
+
+        assertCommandSuccess(addAppointmentCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_duplicateAppointment_throwsCommandException() {
+        Person firstPerson = model.getAddressBook().getPersonList().get(0);
+        UUID personId = firstPerson.getId();
+        AppointmentTime appointmentTime = new AppointmentTime("10/04/2024 2PM-3PM");
+        Appointment appointmentToAdd = new Appointment(personId, appointmentTime);
+        AddAppointmentCommand addAppointmentCommand = new AddAppointmentCommand(INDEX_FIRST_PERSON, appointmentTime);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.addAppointment(appointmentToAdd);
```

Comment:
Is this a duplicate appointment because this function isn't stateless? i.e. the person state from the test above carries to this.

Reply:
To my knowledge the person state from the test above doesn't carry to this, because the original `model` defined in Line 28 doesn't actually get touched.